src/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs

```diff
@@ -449,39 +449,55 @@ private static FileStream OpenStream(int fd, FileAccess access)
         /// </remarks>
         private static void ParseArgumentsIntoList(string arguments, List<string> results)
         {
-            var currentArgument = new StringBuilder();
-            bool inQuotes = false;
-
             // Iterate through all of the characters in the argument string.
             for (int i = 0; i < arguments.Length; i++)
             {
+                while (i < arguments.Length && (arguments[i] == ' ' || arguments[i] == '\t'))
+                    i++;
+
+                if (i == arguments.Length)
+                    break;
+
+                results.Add(GetNextArgument(arguments, ref i));
+            }
+        }
+
+        private static string GetNextArgument(string arguments, ref int i)
+        {
+            var currentArgument = StringBuilderCache.Acquire();
+            bool inQuotes = false;
+
+            while (i < arguments.Length)
+            {
                 // From the current position, iterate through contiguous backslashes.
                 int backslashCount = 0;
-                for (; i < arguments.Length && arguments[i] == '\\'; i++, backslashCount++) ;
+                while (i < arguments.Length && arguments[i] == '\\')
+                {
+                    i++;
+                    backslashCount++;
+                }
+
                 if (backslashCount > 0)
                 {
                     if (i >= arguments.Length || arguments[i] != '"')
```

Comment:
The code would read clearer in my mind if you swapped the if and else and the conditional became a positive:

   `if (i < arguments.Length && arguments[i] == '"')`